Modules/_sre/sre.c

```diff
@@ -1167,13 +1167,21 @@ compile_template(_sremodulestate *module_state,
                  PatternObject *pattern, PyObject *template)
 {
     /* delegate to Python code */
-    PyObject *func = module_state->compile_template;
+    PyObject *func = FT_ATOMIC_LOAD_PTR_RELAXED(module_state->compile_template);
```

Comment:
Let's use `FT_ATOMIC_LOAD_PTR` here.

`FT_ATOMIC_LOAD_PTR_ACQUIRE` would probably also be fine, but I think symmetry between  the load and store is generally a good idea.